sparkctl/cmd/create.go

```diff
@@ -289,9 +289,12 @@ type uploadHandler struct {
 func (uh uploadHandler) uploadToBucket(uploadPath, localFilePath string) (string, error) {
 	fileName := filepath.Base(localFilePath)
 	uploadFilePath := filepath.Join(uploadPath, fileName)
+
 	// Check if exists by trying to fetch metadata
 	reader, err := uh.b.NewRangeReader(uh.ctx, uploadFilePath, 0, 0)
-	reader.Close()
+	if err == nil {
```

Comment:
It's better to do `if reader != nil` instead.

Reply:
I tried it already and did not work. The problem is similar to this one - https://stackoverflow.com/a/16280362 - that reader gets closed with error